contracts/RollupBurnAuction.sol

```diff
@@ -0,0 +1,135 @@
+pragma solidity ^0.5.1;
+import './RollupInterface.sol';
+
+contract RollupBurnAuction {
+
+    // Rollup smart contract address
+    RollupInterface rollupInterface;
+
+    uint32 constant BLOCKS_PER_SLOT = 100;
+    uint32 constant DELTA_BLOCKS_INITIAL_SLOT = 1000;
+
+    // First block where the first era begins
+    uint public genesisBlock;
+
+    // Bid is a structure indicating that
+    // "operator" is willing to burn certain
+    // "amount" of ether in order to forge a slot
+    struct Bid {
+        address payable operator;
+        uint128 amount;
+    }
+
+    struct Auction {
+        uint32 slot;
+        Bid bestBid;
+    }
+
+    address public forger;
+    Auction winner;
```

Comment:
I would save all winners for each slot:
`mapping (uint32 => Auction) winners;`
Then, we can retrieve winner for each slot which can be used as an information outside smart contract:
`function getSlotWinner(uint32 slot) public view returns (address winner)`